crates/iceberg/src/spec/manifest.rs

```diff
@@ -0,0 +1,671 @@
+                column_sizes: value.column_sizes.map(to_i64_entry),
+                value_counts: value.value_counts.map(to_i64_entry),
+                null_value_counts: value.null_value_counts.map(to_i64_entry),
+                nan_value_counts: value.nan_value_counts.map(to_i64_entry),
+                distinct_counts: value.distinct_counts.map(to_i64_entry),
+                lower_bounds: value.lower_bounds.map(to_bytes_entry),
+                upper_bounds: value.upper_bounds.map(to_bytes_entry),
+                key_metadata: value.key_metadata,
+                split_offsets: value.split_offsets,
+                equality_ids: value.equality_ids,
+                sort_order_id: value.sort_order_id,
+            })
+        }
+        pub fn try_into(
+            self,
+            partition_type: &Type,
+            schema: &Schema,
+        ) -> Result<super::DataFile, Error> {
+            let partition = self
+                .partition
+                .try_into(&partition_type)?
+                .and_then(|v| {
+                    if let Literal::Struct(v) = v {
+                        Some(Ok(v))
+                    } else {
+                        Some(Err(Error::new(
+                            ErrorKind::DataInvalid,
+                            "partition value is not a struct",
+                        )))
+                    }
+                })
+                .transpose()?
+                .unwrap_or_default();
+            Ok(super::DataFile {
+                content: self.content.try_into()?,
+                file_path: self.file_path,
+                file_format: self.file_format.parse()?,
+                partition,
+                record_count: self.record_count,
+                file_size_in_bytes: self.file_size_in_bytes,
+                column_sizes: self.column_sizes.map(parse_i64_entry),
+                value_counts: self.value_counts.map(parse_i64_entry),
+                null_value_counts: self.null_value_counts.map(parse_i64_entry),
+                nan_value_counts: self.nan_value_counts.map(parse_i64_entry),
+                distinct_counts: self.distinct_counts.map(parse_i64_entry),
+                lower_bounds: self
+                    .lower_bounds
+                    .map(|v| parse_bytes_entry(v, schema))
+                    .transpose()?,
+                upper_bounds: self
+                    .upper_bounds
+                    .map(|v| parse_bytes_entry(v, schema))
+                    .transpose()?,
+                key_metadata: self.key_metadata,
+                split_offsets: self.split_offsets,
+                equality_ids: self.equality_ids,
+                sort_order_id: self.sort_order_id,
+            })
+        }
+    }
+
+    #[serde_as]
+    #[derive(Serialize, Deserialize)]
+    #[cfg_attr(test, derive(Debug, PartialEq, Eq))]
+    struct BytesEntry {
+        key: i32,
+        #[serde_as(as = "Bytes")]
+        value: Vec<u8>,
```

Comment:
Would it make sense to use `serde_bytes::ByteBuf` here? It is supposed to have an optimized serialization/deserialization.